lms/static/js/commerce/views/receipt_view.js

```diff
@@ -253,7 +253,13 @@ var edx = edx || {};
                 for (var i = 0; i < length; i++) {
                     var line = order.lines[i],
                         attributeValues = _.find(line.product.attribute_values, function (attribute) {
-                            return attribute.name === 'course_key'
+                            // If the attribute has a 'code' property, compare its value, otherwise compare 'name'
+                            var value_to_match = 'course_key';
+                            if (attribute.code) {
+                                return attribute.code === value_to_match;
+                            } else {
```

Comment:
Right, it will need to stay until https://github.com/edx/ecommerce/pull/736 has been deployed to production -- if that ships before this branch, I'll update accordingly -- otherwise we'll have to submit a follow-on PR to clean it up.